internal/charmstore/addentity.go

```diff
@@ -527,6 +527,10 @@ func (s *Store) addCharm(c charm.Charm, p addParams) (err error) {
 		CharmRequiredInterfaces: interfacesForRelations(c.Meta().Requires),
 		SupportedSeries:         c.Meta().Series,
 	}
+	metrics := c.Metrics()
+	if metrics != nil && len(metrics.Metrics) > 0 {
```

Comment:
Technically I think there's no need to check the length field, as c.Metrics should return an empty metrics entry in this case. But perhaps worth being defensive. Or... maybe not, as I guess Metrics might grow another field in the future and then this logic might be incorrect.